contracts/hyphen/token/LPToken.sol

```diff
@@ -85,24 +99,49 @@ contract LPToken is
         return _exists(_tokenId);
     }
 
-    function tokenURI(uint256 tokenId)
+    function supportsInterface(bytes4 interfaceId)
         public
         view
         virtual
-        override(ERC721Upgradeable, ERC721URIStorageUpgradeable)
-        returns (string memory)
+        override(ERC721Upgradeable, ERC721EnumerableUpgradeable)
+        returns (bool)
     {
-        return ERC721URIStorageUpgradeable.tokenURI(tokenId);
+        return super.supportsInterface(interfaceId);
     }
 
-    function supportsInterface(bytes4 interfaceId)
+    function tokenURI(uint256 tokenId)
         public
         view
         virtual
-        override(ERC721Upgradeable, ERC721EnumerableUpgradeable)
-        returns (bool)
+        override(ERC721Upgradeable, ERC721URIStorageUpgradeable)
+        returns (string memory)
     {
-        return super.supportsInterface(interfaceId);
+        address tokenAddress = tokenMetadata[tokenId].token;
+        require(svgHelpers[tokenAddress] != ISvgHelper(address(0)), "ERR__SVG_HELPER_NOT_REGISTERED");
+
+        string memory svgData = svgHelpers[tokenAddress].getTokenSvg(
+            tokenId,
+            tokenMetadata[tokenId].suppliedLiquidity,
+            ILiquidityProviders(liquidityProvidersAddress).totalReserve(tokenAddress)
+        );
+
+        string memory json = Base64.encode(
+            bytes(
+                string(
+                    abi.encodePacked(
+                        '{"name": "',
+                        name(),
+                        '", "description": "',
+                        // TODO: Check description and opensea attributes
+                        "",
```

Comment:
No description here? We can add
"NFTs representing your position on Hyphen Liquidity Pools."